Use the Spring Initializr

Buzzardo opened the pull request:
I rewrote the document and the code (mostly changing package names) to use the Spring Initializr. This particular example involved modifying both the Maven and the Gradle build files, which added quite a bit to the document.

Commit:
Use the Spring Initializr

I rewrote the document and the code (mostly changing package names) to use the Spring Initializr. This particular example involved modifying both the Maven and the Gradle build files, which added quite a bit to the document.

dsyer commented on complete/application/build.gradle:
```diff
@@ -1,30 +1,20 @@
-    testCompile('org.springframework.boot:spring-boot-starter-test')
+	implementation 'org.springframework.boot:spring-boot-starter-actuator'
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	compile project(':library')
```
I'm not a Gradler, but why is this one still "compile"? isn't that out-moded?

Buzzardo replied:
I couldn't get the implementation way to work, so I went back to the compile way. If someone has another way, I'd love to see it.

dsyer replied:
Calling @wilkinsona ...

wilkinsona replied:
It should work with `library` and does so for me:

```
[$]: ./gradlew build
Downloading https://services.gradle.org/distributions/gradle-4.6-bin.zip
......................................................................
Unzipping /Users/awilkinson/.gradle/wrapper/dists/gradle-4.6-bin/4jp4stjndanmxuerzfseyb6wo/gradle-4.6-bin.zip to /Users/awilkinson/.gradle/wrapper/dists/gradle-4.6-bin/4jp4stjndanmxuerzfseyb6wo
Set executable permissions for: /Users/awilkinson/.gradle/wrapper/dists/gradle-4.6-bin/4jp4stjndanmxuerzfseyb6wo/gradle-4.6/bin/gradle
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :application:test 
2019-09-27 09:04:12.806  INFO 14694 --- [       Thread-5] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'


BUILD SUCCESSFUL in 37s
9 actionable tasks: 9 executed
[$]: git diff HEAD
diff --git a/complete/application/build.gradle b/complete/application/build.gradle
index 28336a1..b452658 100644
--- a/complete/application/build.gradle
+++ b/complete/application/build.gradle
@@ -15,6 +15,6 @@ repositories {
 dependencies {
        implementation 'org.springframework.boot:spring-boot-starter-actuator'
        implementation 'org.springframework.boot:spring-boot-starter-web'
-       compile project(':library')
+       implementation project(':library')
        testImplementation 'org.springframework.boot:spring-boot-starter-test'
 }
```

dsyer commented on initial/application/build.gradle:
```diff
@@ -1,30 +1,19 @@
-    testCompile('org.springframework.boot:spring-boot-starter-test')
+	implementation 'org.springframework.boot:spring-boot-starter-actuator'
+	implementation 'org.springframework.boot:spring-boot-starter-web'
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
```
Missing dependency on library? I guess that's OK for the initial?

Buzzardo replied:
(I thought I answered this, but I don't see it. Sorry if there are two answers.)

I thought I would use only what the Initializr generated in the initial project and have the complete project contain the bits as they look after editing.

Commit:
Updated build.gradle to use "implementation" rather than "compile"

Dave Syer pointed out that "compile" is outdated and that it should be "implementation" and Andy Wilkinson figured out how to make that work. (The odd thing is that I had tried it and couldn't get it to work. I must have had some other error, too.) I made the change Andy suggested, and it works now. Thanks, Dave and Andy. :)
